ISX/ISX/Constants.swift

```diff
@@ -34,20 +46,103 @@ struct Constants {
     
 //      PRAGMA MARK: Row heights
     static let tableViewRowHeight:CGFloat = 85
 
 //      PRAGMA MARK: Cells
     static let progressBarCellHeight:CGFloat = 100
     static let requestCellHeight:CGFloat = 130
+    static let requestProductCellSize: CGFloat = 155
+    static let requestLabelCellSize: CGFloat = 30
+    static let requestLocationCellSize: CGFloat = 155
+    static let requestFavoriteCellSize: CGFloat = 50
+    static let requestExtraProductInfoCellSize: CGFloat = 400
     static let searchBarCellHeight:CGFloat = 60
+
+    //PRAGMA MARK: Firebase
+    static let firebaseFavoriteTable = "favorite"
+    private static let firebaseDataroot = "dataroot"
+    static let firebaseProductsTable = "products"
+    static let firebaseFlightsTable = "flights"
+    static let firebaseProductGroupsTable = "productGroups"
+    static let firebaseRequestsTable = "requests"
+    private static let keepFirebaseSynced = true
+
     
 //      PRAGMA MARK: Calculate miles
     static let multiplierFactorMiles = 400
+    
+    //PRAGMA MARK: Buttons
+    static let popupButtonHeight = 60
+    
+    //PRAGMA MARK: Reusable identifiers
+    static let progressbarCell: String = "progressCell"
+    static let CCRequestCell: String = "CustomRequestCell"
+    static let CCrequestProductInfo: String = "requestProductInfoCell"
+    static let CCLocationLabel: String = "requestLocationLabelCell"
+    static let CCLocationInfo: String = "requestLocationInfoCell"
+    static let CCFavoritesLabel: String = "requestFavoritesLabelCell"
+    static let CCfavoritesInfo: String = "favoritesInRequestCell"
+    static let CCExtraProductDetail: String = "requestExtraProductDetailCell"
+    static let CCExtraProductLabel: String = "requestExtraProductLabelCell"
+    
+    //PRAGMA MARK: Methods
+    
+    static func isFirebaseSynced() -> Bool { return keepFirebaseSynced }
+    
+    private static func getRootRef() -> DatabaseReference { return Database.database().reference(withPath: firebaseDataroot) }
+    
+    static func getProductRef() -> DatabaseReference {
+        let databaseRootRef = getRootRef()
+        return databaseRootRef.child(firebaseProductsTable)
+    }
+    
+    static func getFavoriteRef() -> DatabaseReference {
+        let databaseRootRef = getRootRef()
+        return databaseRootRef.child(firebaseFavoriteTable)
+    }
+    
+    static func getRequestRef() -> DatabaseReference {
+        let databaseRootRef = getRootRef()
+        return databaseRootRef.child(firebaseRequestsTable)
+    }
+    
+    static func getFlightsRef() -> DatabaseReference {
+        let databaseRootRef = getRootRef()
+        return databaseRootRef.child(firebaseFlightsTable)
+    }
+    
+    static func getProductGroupRef() -> DatabaseReference {
+        let databaseRootRef = getRootRef()
+        return databaseRootRef.child(firebaseProductGroupsTable)
+    }
+    
+    static func setRequest(requestLatestId: Int,productId: Int, customerChairNumber: String,
```

Comment:
why is this in constants?

Reply:
It can also be used in ProductInfoController, but not part of my hotfix, so not touching that yet.